llvm/CMakeLists.txt

```diff
@@ -291,7 +291,7 @@ if(LLVM_CCACHE_BUILD)
       set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_PROGRAM})
     else()
       if(LLVM_CCACHE_MAXSIZE OR LLVM_CCACHE_DIR OR
-         NOT LLVM_CCACHE_PARAMS MATCHES "CCACHE_CPP2=yes CCACHE_HASHDIR=yes")
+         NOT LLVM_CCACHE_PARAMS MATCHES "CCACHE_CPP2=yes CCACHE_HASHDIR=yes CCACHE_SLOPPINESS=pch_defines,time_macros")
```

Comment:
If your Windows question was for this, I it will cause no cache hits because nothing sets `CCACHE_SLOPPINESS=pch_defines,time_macros`.

Workarounds could be 
1. `env` command from MSYS
2. `set` command from Windows itself (only a single variable per invocation and does not support specifying a command on the same line but can be  chained with `&&`): https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/set_1
3. emit a wrapper around `ccache` that sets those enviornment variables
4. test & abort if ccache does not use the required sloppiness option

Reply:
As written this will end up with compilation commands like
`"/usr/local/bin/cmake -E env CCACHE_CPP2=yes CCACHE_HASHDIR=yes CCACHE_SLOPPINESS=pch_defines,time_macros -- /usr/bin/ccache" /usr/bin/clang++ -DGTEST_H(..)`
and fail with
`/bin/sh: 1: /usr/local/bin/cmake -E env CCACHE_CPP2=yes CCACHE_HASHDIR=yes CCACHE_SLOPPINESS=pch_defines,time_macros -- /usr/bin/ccache: not found`
due to `launcher` getting put in quote marks. Removing the quotemarks from the last two lines in the patch above does not seem to make a difference, is there some way we could get around it?

Reply:
Ah ok, we just need to be careful with the quoting. This works:
```
+    set(launcher ${CMAKE_COMMAND} -E env ${launcher_env} ${CCACHE_PROGRAM})
+    set(CMAKE_C_COMPILER_LAUNCHER ${launcher})
+    set(CMAKE_CXX_COMPILER_LAUNCHER ${launcher})
```

Reply:
See the new commit for a working solution.